### ISSUE TITLE
Unable to fetch content. Reason: "certificate verify failed"

### ISSUE THREAD

**Issue by xenoexclusive** [paritytech/parity#7961]
> I'm running:
>
> - **Which Parity version?**: 1.8.10
> - **Which operating system?**: Ubuntu/Arm64
> - **How installed?**: Binaries
> - **Are you fully synchronized?**: Yes
> - **Which network are you connected to?**: ethereum foundation
> - **Did you try to restart the node?**: Yes

I access my node via SSH into my odroid, if that makes a difference.

Upon start of parity, the first sign of the error appears during initialisation with

> Failed to auto-update latest ETH price: Fetch(Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: > Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: 
> "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: 
> Some("https://api.etherscan.io/api?module=stats&action=ethprice") }))

Then upon attempting to access the browser wallet, web3 websites fail to load and in wallet images (such as application icons etc) will not load with the following similar errors.

> 2018-02-21 21:05:06  Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: 
> Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: 
> "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: 
> Some("https://raw.githubusercontent.com/ethcore/dapp-assets/ec6138115d0e1f45258969cd90b3b274e0ff2258/dapps/earth-64x64.jpg") })
> 2018-02-21 21:05:06  Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: 
> Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: 
> "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: 
> Some("https://raw.githubusercontent.com/ethcore/dapp-assets/88a17093c5ec97eead622e3944724ac18d2931fc/dapps/register-64x64.jpg") })
> 2018-02-21 21:05:06  Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: 
> Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: 
> "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: 
> Some("https://raw.githubusercontent.com/ethcore/dapp-assets/cdd6ac4f1e2f11619bed72a53ae71217dffe19ad/dapps/stack-64x64.png") })

etc etc with more errors depending on how many images for example are attempting to be loaded.

Basically renders all web3 content inaccessible.

**Comment by mafonso** [comment]
I'm experiencing the same on 1.9.3 beta on Linux/armv7 (raspberry pi) 
Not observed with the same version on Darwin 

**Parity version:** version Parity/v1.9.3-beta-804ddfe-20180219/arm-linux-gnu/rustc1.23.0
**OS:** armv7l GNU/Linux

```
Feb 25 10:23:15 rpi-master parity[23706]: 2018-02-25 10:23:15 UTC Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: Some("https://raw.githubusercontent.com/ethcore/dapp-assets/b88e983abaa1a6a6345b8d9448c15b117ddb540e/dapps/hex-64x64.jpg") })
Feb 25 10:23:20 rpi-master parity[23706]: 2018-02-25 10:23:20 UTC Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: Some("https://raw.githubusercontent.com/ethcore/console/3ea0dbfefded359ccdbea37bc4cf350c0aa16948/console.jpeg") })
Feb 25 10:23:22 rpi-master parity[23706]: 2018-02-25 10:23:22 UTC Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: Some("https://raw.githubusercontent.com/ethcore/dapp-assets/b88e983abaa1a6a6345b8d9448c15b117ddb540e/dapps/register-64x64.jpg") })
Feb 25 10:23:24 rpi-master parity[23706]: 2018-02-25 10:23:24 UTC Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: Some("https://raw.githubusercontent.com/ethcore/dapp-assets/b88e983abaa1a6a6345b8d9448c15b117ddb540e/dapps/interlock-64x64.png") })
Feb 25 10:23:25 rpi-master parity[23706]: 2018-02-25 10:23:25 UTC Unable to fetch content: Fetch(Error { kind: Io(Error { repr: Custom(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336134278, library: "SSL routines", function: "ssl3_get_server_certificate", reason: "certificate verify failed", file: "s3_clnt.c", line: 1269 }])) }) }), url: Some("https://raw.githubusercontent.com/ethcore/dapp-assets/b88e983abaa1a6a6345b8d9448c15b117ddb540e/dapps/coins-64x64.jpg") })
```

**Comment by xenoexclusive** [comment]
Likely to be something to do with the arm binaries at a guess!

**Comment by 5chdn** [comment]
Probably outdated certificates on Ubuntu? I manually checked the certs and they are valid. Not a Parity issue.

**Comment by binarybana** [comment]
I believe this an error upstream on ARM platforms (or at least my raspberry pi). See https://github.com/seanmonstar/reqwest/issues/276 for example.

**Comment by xenoexclusive** [comment]
You're absolutely right, it does. Do you recommend the workaround provided or do you think a future update will resolve these issues?

Apologies, I'm not really technically experienced in this stuff at all.

**Comment by binarybana** [comment]
I'm just using the env var myself. Not too keen to go try and hack the cross compiler.

**Comment by xenoexclusive** [comment]
I have no idea what that means :D

**Comment by xenoexclusive** [comment]
Just to confirm, because I'm truly an idiot, what exactly am I putting in place of './your-binary'?

`$ SSL_CERT_DIR=/etc/ssl/certs ./your-binary`

**Comment by xenoexclusive** [comment]
Just to confirm, I fixed this by starting parity with the line:

`$ SSL_CERT_DIR=/etc/ssl/certs /usr/bin/parity`
